django/test/runner.py

```diff
@@ -52,19 +58,260 @@ def printErrorList(self, flavour, errors):
+
+    This helper lives at module-level because of the multiprocessing module's
+    requirements.
+    """
+
+    global _worker_id
+
+    with counter.get_lock():
+        counter.value += 1
+        _worker_id = counter.value
+
+    for alias in connections:
+        connection = connections[alias]
+        settings_dict = connection.creation.get_test_db_clone_settings(_worker_id)
+        connection.settings_dict = settings_dict
+        connection.close()
+
+
+def _run_subsuite(args):
+    """
+    Run a suite of tests with a RemoteTestRunner and return a RemoteTestResult.
+
+    This helper lives at module-level and its arguments are wrapped in a tuple
+    because of the multiprocessing module's requirements.
+    """
+
+    subsuite_index, subsuite, failfast = args
+    runner = RemoteTestRunner(failfast=failfast)
+    result = runner.run(subsuite)
+    return subsuite_index, result.events
+
+
+class ParallelTestSuite(unittest.TestSuite):
+    """
+    Run a series of tests in parallel in several processes.
+
+    While the unittest module's documentation implies that orchestrating the
+    execution of tests is the responsibility of the test runner, in practice,
+    it appears that TestRunner classes are more concerned with formatting and
+    displaying test results.
+
+    Since there are fewer use cases for customizing TestSuite than TestRunner,
+    implementing parallelization at the level of the TestSuite improves
+    interoperability with existing custom test runners. A single instance of a
+    test runner can still collect results from all tests without being aware
+    that they have been run in parallel.
+    """
+
+    # In case someone wants to modify these in a subclass.
+    init_worker = _init_worker
+    run_subsuite = _run_subsuite
+
+    def __init__(self, suite, processes, failfast=False):
+        self.subsuites = partition_suite_by_case(suite)
+        self.processes = processes
+        self.failfast = failfast
+        super(ParallelTestSuite, self).__init__()
+
+    def run(self, result):
+        # Distribute test cases across workers. Return an identifier of each
+        # test case with its result in order to be able to use imap_unordered
+        # and show results as soon as they're available.
+
+        # To minimize pickling errors when getting results from workers:
+        # - pass back numeric indexes in self.subsuites instead of tests
+        # - make tracebacks pickleable with tblib
+        # Errors may still occur for dynamically created exception classes
+        # like Model.DoesNotExist. Pickling them seems very difficult.
+
+        # Inner import to prevent a hard dependency on tblib.
+        from tblib import pickling_support
+        pickling_support.install()
+
+        counter = multiprocessing.Value(ctypes.c_int, 0)
+        pool = multiprocessing.Pool(
+            processes=self.processes,
+            initializer=type(self).init_worker,
```

Comment:
I need a plain function here. `self.init_worker` would give me a bound method.

I think the issue you raised above is because `type(self).init_worker` is an unbound method on Python 2. I didn't test on Python 2.